### PR TITLE
feature/global_det_date_ticks

### PR DESCRIPTION
**Pull request by malloryprow**
## Pull Request Testing ##

1.  Add hour information on date tick labels

- [ ] Describe testing already performed for this Pull Request:</br>
> I ran the affected jobs and reviewed the images.

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

> **Set-up**
> 1. Clone my fork and checkout branch feature/global_det_date_ticks
> 2. ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix
> 
> **plots**
> 3. Run dev/drivers/scripts/plots/global_det/jevs_global_det_atmos_grid2grid_precip_plots_31days.sh, dev/drivers/scripts/plots/global_det/jevs_global_det_atmos_grid2grid_precip_plots_90days.sh, dev/drivers/scripts/plots/global_det/jevs_global_det_atmos_grid2grid_pres_levs_plots_31days.sh, dev/drivers/scripts/plots/global_det/jevs_global_det_atmos_grid2grid_pres_levs_plots_90days.sh, and dev/drivers/scripts/plots/global_det/jevs_global_det_atmos_headline_plots.sh. 

- [ ] Has the code been checked to ensure that no errors occur during the execution? **Yes**

- [ ] Do these updates/additions include sufficient testing updates? **Yes**

- [ ] Please complete this pull request review by **11/21/2023**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
I have cloned your fork and checked out branch `feature/global_det_date_ticks`.

**Comment by ShelleyMelchior-NOAA** [comment]
The 5 plots processes have completed. I did not untar to look at the ticks.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_det/`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_det_atmos_*_plots*.o93482*`

**Comment by AliciaBentley-NOAA** [comment]
@ShelleyMelchior-NOAA @malloryprow Sorry for asking this, but the other email thread about global_ens got a little hard to follow. Is Mallory adding the global_ens changes to this PR and are we testing them here?

**Comment by ShelleyMelchior-NOAA** [comment]
I opted to only test global_det changes for now. If Binbin is doing them, that's fine. I did not ask Mallory to make the global_ens changes, nor did I introduce them yet, b/c I did not want duplicate efforts.

**Comment by AliciaBentley-NOAA** [comment]
@ShelleyMelchior-NOAA Thanks. This was the last thing I saw about this PR in that email thread, so I thought it might be happening here: _"Mallory, since you are attentive on this matter and have helped to diagnose the problem, I can work it into the global_det PR. Cool?"_ It's totally fine if it's not in this PR, but wanted to make sure everyone was on the same page.

**Comment by malloryprow** [comment]
> The 5 plots processes have completed. I did not untar to look at the ticks. COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_det/` logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_det_atmos_*_plots*.o93482*`

Going to start reviewing them now!

**Comment by ShelleyMelchior-NOAA** [comment]
> @ShelleyMelchior-NOAA Thanks. This was the last thing I saw about this PR in that email thread, so I thought it might be happening here: _"Mallory, since you are attentive on this matter and have helped to diagnose the problem, I can work it into the global_det PR. Cool?"_ It's totally fine if it's not in this PR, but wanted to make sure everyone was on the same page.

Correct. I did write that. But I didn't get confirmation and then Binbin swooped in. I was also attending to the overarching PR re: evs_ver so I was allowing space and time for the global_ens situation of who will do the work to get resolved. I understand it is not fully resolved at this moment. I can try to figure that out soon.

**Comment by malloryprow** [comment]
The plots don't look like what I expected. I see HOMEevs=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/EVS. Is that where it should be pointing?

**Comment by ShelleyMelchior-NOAA** [comment]
Gah! You are correct. My symlink is pointing to Ho-Chun's PR. I will correct that and re-run. Thankfully your 5 plots scripts are quick. My brain is not firing on its best cylinders today.

**Comment by malloryprow** [comment]
All good! The jobs are quick to run.

**Comment by ShelleyMelchior-NOAA** [comment]
The re-run is underway. The plots will write to the same COMOUT noted above.
```
cbqs01: 
                                                                                                   Req'd  Req'd   Elap
Job ID                         Username        Queue           Jobname         SessID   NDS  TSK   Memory Time  S Time
------------------------------ --------------- --------------- --------------- -------- ---- ----- ------ ----- - -----
93491517.cbqs01                shelley.melchi* dev             jevs_global_de*    31734    1   128    --  01:00 R 00:00
93491518.cbqs01                shelley.melchi* dev             jevs_global_de*   203825    1   128    --  01:15 R 00:00
93491519.cbqs01                shelley.melchi* dev             jevs_global_de*   100916    1   128    --  00:30 R 00:00
93491520.cbqs01                shelley.melchi* dev             jevs_global_de*   129606    1   128    --  00:45 R 00:00
93491521.cbqs01                shelley.melchi* dev             jevs_global_de*   232248    1     1   35gb 00:10 R 00:00
```

**Comment by malloryprow** [comment]
Oh shoot. Okay we may need to clear COMOUT due to the restart capabilties.

**Comment by ShelleyMelchior-NOAA** [comment]
ok, I will qdel and start fresh for COMOUT.

**Comment by ShelleyMelchior-NOAA** [comment]
Ok, qdel'd, COMOUT rm'd, 5 plots drivers resubmitted.

**Comment by ShelleyMelchior-NOAA** [comment]
The re-submissions have completed.

**Comment by malloryprow** [comment]
I'm not sure what is going on but the plots still don't look right. I checked the headline and pres_levs last 31 day tar files, and they still look like the old images.

**Comment by ShelleyMelchior-NOAA** [comment]
hmm, let me triple check my set up ...

**Comment by ShelleyMelchior-NOAA** [comment]
I found the problem in my set up. The plots are re-running.

**Comment by ShelleyMelchior-NOAA** [comment]
I have pulled in the global_ens changes. I will run the appropriate global_ens processes to test these changes.

**Comment by malloryprow** [comment]
YAYYYYYY

image
image
image
image

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow and @ShelleyMelchior-NOAA Thanks so much! I'd like to confirm with @GeoffManikin-NOAA , @LoganDawson-NOAA , and Jason if the UTC time needs to be in front of the date or not and will report back. That's how it's written in AMS journals (time before date), and I'd like to get their take on it quickly before we say this is complete. Thanks!

**Comment by malloryprow** [comment]
I switched gefs_number and gefs_mbrs in the two prep scripts, which is why they failed but fixed now!

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow How hard would it be to move the time in front of the date and retest the plotting scripts? This is the AMS journal convention (time and then date, see first paragraph and first image caption [in this MWR paper](https://journals.ametsoc.org/view/journals/mwre/147/5/mwr-d-18-0453.1.xml)), and it would help our EVS graphics be publication quality to reorder it. If it's not too much more work to reorder it, we'd like this change to be made. What do you think?

**Comment by malloryprow** [comment]
> @malloryprow How hard would it be to move the time in front of the date and retest the plotting scripts? This is the AMS journal convention (time and then date, see first paragraph and first image caption [in this MWR paper](https://journals.ametsoc.org/view/journals/mwre/147/5/mwr-d-18-0453.1.xml)), and it would help our EVS graphics be publication quality to reorder it. If it's not too much more work to reorder it, we'd like this change to be made. What do you think?

Just pushed the changes

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow Thank you so much! That's a small change that will make a big difference to the people who might put EVS images in papers! 

@ShelleyMelchior-NOAA If you could retest the plotting scripts one more time with Mallory's latest changes to show that the  images were updated, that would be amazing. Thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
I will retest the plots and the global_ens prep. I will clean out COMOUT for plots. I will pull in the new changes. I am writing out everything explicitly b/c I am making mistakes. Trying to be deliberate.

**Comment by malloryprow** [comment]
Do what you need to do! :)

**Comment by ShelleyMelchior-NOAA** [comment]
Everything is running now! All log files are in: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/`
COMOUTs are:
`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens/`
`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/`
And the same COMOUT for global_det plots as before.

**Comment by malloryprow** [comment]
YAYYYYY, round 2

image
image
image
image

**Comment by AliciaBentley-NOAA** [comment]
@malloryprow @ShelleyMelchior-NOAA YAY! Fantastic job! I approve these updated graphics! Did the global_ens testing finish with the corrected variable names or are we stilling waiting on that to round out this PR?

**Comment by ShelleyMelchior-NOAA** [comment]
global_ens is almost finished running. Mallory is putting in fixes for the problem Binbin found.
global_ens stats COMOUT: ` /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/global_ens/naefs.20231115`
global_ens prep COMOUT: `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/global_ens`

**Comment by ShelleyMelchior-NOAA** [comment]
Shall I run the impacted global_ens plots scripts?

**Comment by StevenSimon-NOAA** [comment]
The naefs prep files (cmce_bc and gefs_bc prep files) and headline prep files (cmce and gefs) are fully accounted for by expected quantities.

**Comment by ShelleyMelchior-NOAA** [comment]
I kicked off the following global_ens plots processes:
```
-rw-r--r-- 1 shelley.melchior emc   776228 Nov 17 18:53 jevs_global_ens_atmos_gefs_precip_spatial_plots.o93502897
-rw-r--r-- 1 shelley.melchior emc  2208215 Nov 17 18:56 jevs_global_ens_atmos_gefs_sea_ice_past31days_plots.o93503069
-rw-r--r-- 1 shelley.melchior emc  2037810 Nov 17 18:56 jevs_global_ens_atmos_gefs_snowfall_past90days_plots.o93503085
-rw-r--r-- 1 shelley.melchior emc  2334266 Nov 17 18:57 jevs_global_ens_atmos_gefs_sea_ice_past90days_plots.o93503071
-rw-r--r-- 1 shelley.melchior emc  2056306 Nov 17 18:57 jevs_global_ens_atmos_gefs_precip_past31days_plots.o93502889
-rw-r--r-- 1 shelley.melchior emc   957574 Nov 17 18:57 jevs_global_ens_atmos_naefs_grid2grid_past31days_plots.o93503179
-rw-r--r-- 1 shelley.melchior emc   958459 Nov 17 18:57 jevs_global_ens_atmos_gefs_sst_past31days_plots.o93503096
-rw-r--r-- 1 shelley.melchior emc  1259110 Nov 17 18:58 jevs_global_ens_atmos_naefs_grid2grid_past90days_plots.o93503182
-rw-r--r-- 1 shelley.melchior emc  2117194 Nov 17 18:58 jevs_global_ens_atmos_gefs_snowfall_past31days_plots.o93503083
-rw-r--r-- 1 shelley.melchior emc   955895 Nov 17 18:58 jevs_global_ens_atmos_gefs_sst_past90days_plots.o93503097
-rw-r--r-- 1 shelley.melchior emc  1063364 Nov 17 18:58 jevs_global_ens_atmos_naefs_grid2obs_past31days_plots.o93503183
-rw-r--r-- 1 shelley.melchior emc  1975305 Nov 17 18:58 jevs_global_ens_atmos_gefs_precip_past90days_plots.o93502894
-rw-r--r-- 1 shelley.melchior emc  1210498 Nov 17 18:58 jevs_global_ens_atmos_naefs_grid2obs_past90days_plots.o93503184
[clogin02 /lfs/h2/emc/ptmp/shelley.melchior/EVS_out]$
```

**Comment by StevenSimon-NOAA** [comment]
All large naefs stats files are fully accounted for by expected quantity. While the small naefs stats files are not fully accounted for by expected quantity, they do match those in emc.vpppg. So this reflects the condition of the emc.vpppg prep and stats archive and the impacts that WCOSS2 issues and dev-prod switches have had on it. There are no apparent errors in the accompanying naefs stats logfiles.

**Comment by ShelleyMelchior-NOAA** [comment]
The tar files are built. 
```
$ ll /lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/global_ens/atmos.20231115
total 1431384
-rw-r--r-- 1 shelley.melchior emc 274698240 Nov 17 19:03 evs.plots.global_ens.atmos.gefs.precip.past31days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc 342384640 Nov 17 19:21 evs.plots.global_ens.atmos.gefs.precip.past90days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc   2560000 Nov 17 19:02 evs.plots.global_ens.atmos.gefs.precip_spatial.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc  55490560 Nov 17 18:56 evs.plots.global_ens.atmos.gefs.sea_ice.past31days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc  63846400 Nov 17 18:57 evs.plots.global_ens.atmos.gefs.sea_ice.past90days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc 289280000 Nov 17 19:03 evs.plots.global_ens.atmos.gefs.snowfall.past31days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc 304025600 Nov 17 19:15 evs.plots.global_ens.atmos.gefs.snowfall.past90days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc   7936000 Nov 17 18:57 evs.plots.global_ens.atmos.gefs.sst.past31days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc   8560640 Nov 17 18:58 evs.plots.global_ens.atmos.gefs.sst.past90days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc  12687360 Nov 17 18:58 evs.plots.global_ens.atmos.naefs.grid2grid.past31days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc  12748800 Nov 17 18:59 evs.plots.global_ens.atmos.naefs.grid2grid.past90days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc  45864960 Nov 17 19:01 evs.plots.global_ens.atmos.naefs.grid2obs.past31days.v20231115.tar
-rw-r--r-- 1 shelley.melchior emc  45639680 Nov 17 19:07 evs.plots.global_ens.atmos.naefs.grid2obs.past90days.v20231115.tar
```

How do the log files look? @BinbinZhou-NOAA 
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_global_ens_atmos_*plots.o9350*`

**Comment by malloryprow** [comment]
The tar files look good!

**Comment by StevenSimon-NOAA** [comment]
> The tar files look good!

And the expected plot png quantities per tar file are fully accounted for.

**Comment by StevenSimon-NOAA** [comment]
After correcting for cat duplication, the generated plots logfiles in this PR are all smaller in size than their emc.vpppg counterparts today. Thus, I think we can conclude that the cat duplication error has been successfully corrected.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All changes look good. All tests yield expected results. This PR is approved for merge.
